patches/base-metrics-histogram_functions.h.patch

```diff
@@ -0,0 +1,20 @@
+diff --git a/base/metrics/histogram_functions.h b/base/metrics/histogram_functions.h
+index 18f460cc992ddac3669f514beee5a66878970653..b5d3637fec97acf90c2ca89138a53f3291068b95 100644
+--- a/base/metrics/histogram_functions.h
++++ b/base/metrics/histogram_functions.h
+@@ -75,6 +75,7 @@ void UmaHistogramEnumeration(const std::string& name, T sample) {
+   static_assert(static_cast<uintmax_t>(T::kMaxValue) <=
+                     static_cast<uintmax_t>(INT_MAX) - 1,
+                 "Enumeration's kMaxValue is out of range of INT_MAX!");
++  if (false)
+   DCHECK_LE(static_cast<uintmax_t>(sample),
+             static_cast<uintmax_t>(T::kMaxValue));
```

Comment:
shouldn't we be fixing `kMaxValue` instead?

Reply:
sorry, it's a negative value so shouldn't we fix the mapping?

Reply:
also shouldn't we be returning early or something if the value is negative? This is going to eventually write the wrong value as an overflow

Reply:
where does this get triggered from?

Reply:
The main place I noticed it is when doing a session restore on startup:

```[55396:40808:0620/200106.219:FATAL:histogram_functions.h(95)] Check failed: static_cast<uintmax_t>(sample) <= static_cast<uintmax_t>(T::kMaxValue) (18446744073709551611 vs. 29)
        base::debug::CollectStackTrace [0x00007FF89EF49212+18] (C:\work\cr127\brave-browser\src\base\debug\stack_trace_win.cc:333)
        base::debug::StackTrace::StackTrace [0x00007FF89EF2A5ED+61] (C:\work\cr127\brave-browser\src\base\debug\stack_trace.cc:237)
        logging::LogMessage::Flush [0x00007FF89EDEDBD4+244] (C:\work\cr127\brave-browser\src\base\logging.cc:715)
        logging::LogMessage::~LogMessage [0x00007FF89EDEDA79+25] (C:\work\cr127\brave-browser\src\base\logging.cc:703)
        logging::`anonymous namespace'::DCheckLogMessage::~DCheckLogMessage [0x00007FF89EDC5210+144] (C:\work\cr127\brave-browser\src\base\check.cc:162)
        logging::CheckError::~CheckError [0x00007FF89EDC4B0C+44] (C:\work\cr127\brave-browser\src\base\check.cc:344)
        base::UmaHistogramEnumeration<PageActionIconType> [0x00007FFFFA205664+196] (C:\work\cr127\brave-browser\src\base\metrics\histogram_functions.h:94)
        PageActionIconController::RecordIndividualMetrics [0x00007FFFFA204EDD+221] (C:\work\cr127\brave-browser\src\chrome\browser\ui\views\page_action\page_action_icon_controller.cc:470)
        PageActionIconController::OnPageActionIconViewShown [0x00007FFFFA204C1C+700] (C:\work\cr127\brave-browser\src\chrome\browser\ui\views\page_action\page_action_icon_controller.cc:362)
        PageActionIconView::SetVisible [0x00007FFFFA1FE8C5+629] (C:\work\cr127\brave-browser\src\chrome\browser\ui\views\page_action\page_action_icon_view.cc:327)
        SpeedreaderIconView::UpdateImpl [0x00007FFFF868CA11+305] (C:\work\cr127\brave-browser\src\brave\browser\ui\views\speedreader\speedreader_icon_view.cc:69)
        PageActionIconView::Update [0x00007FFFFA1FE116+86] (C:\work\cr127\brave-browser\src\chrome\browser\ui\views\page_action\page_action_icon_view.cc:254)
        PageActionIconController::UpdateAll [0x00007FFFFA204329+73] (C:\work\cr127\brave-browser\src\chrome\browser\ui\views\page_action\page_action_icon_controller.cc:299)
        LocationBarView::Update [0x00007FFFFA22B136+102] (C:\work\cr127\brave-browser\src\chrome\browser\ui\views\location_bar\location_bar_view.cc:930)
        BraveLocationBarView::Update [0x00007FFFF86B50CA+154] (C:\work\cr127\brave-browser\src\brave\browser\ui\views\location_bar\brave_location_bar_view.cc:208)
        ToolbarView::Update [0x00007FFFFA18DF35+37] (C:\work\cr127\brave-browser\src\chrome\browser\ui\views\toolbar\toolbar_view.cc:554)
        BraveToolbarView::Update [0x00007FFFF86BD51F+15] (C:\work\cr127\brave-browser\src\brave\browser\ui\views\toolbar\brave_toolbar_view.cc:338)
        Browser::OnActiveTabChanged [0x00007FFFFA39086D+429] (C:\work\cr127\brave-browser\src\chrome\browser\ui\browser.cc:2755)
        Browser::OnTabStripModelChanged [0x00007FFFFA390024+356] (C:\work\cr127\brave-browser\src\chrome\browser\ui\browser.cc:1411)
        BraveBrowser::OnTabStripModelChanged [0x00007FFFF85A60E5+21] (C:\work\cr127\brave-browser\src\brave\browser\ui\brave_browser.cc:161)
        TabStripModel::OnChange [0x00007FFFFA330F8C+588] (C:\work\cr127\brave-browser\src\chrome\browser\ui\tabs\tab_strip_model.cc:393)
        TabStripModel::InsertTabAtIndexImpl [0x00007FFFFA341441+961] (C:\work\cr127\brave-browser\src\chrome\browser\ui\tabs\tab_strip_model.cc:2545)
        TabStripModel::InsertTabAtImpl [0x00007FFFFA33074B+539] (C:\work\cr127\brave-browser\src\chrome\browser\ui\tabs\tab_strip_model.cc:1984)
        TabStripModel::InsertWebContentsAt [0x00007FFFFA33022E+190] (C:\work\cr127\brave-browser\src\chrome\browser\ui\tabs\tab_strip_model.cc:300)
        chrome::AddRestoredTab_ChromiumImpl [0x00007FFFFA6EB3D7+567] (C:\work\cr127\brave-browser\src\chrome\browser\ui\browser_tabrestore.cc:244)
        chrome::AddRestoredTab [0x00007FFFFA6EC09F+351] (C:\work\cr127\brave-browser\src\brave\chromium_src\chrome\browser\ui\browser_tabrestore.cc:62)
        SessionRestoreImpl::RestoreTab [0x00007FFFF94D45ED+941] (C:\work\cr127\brave-browser\src\chrome\browser\sessions\session_restore.cc:923)
        SessionRestoreImpl::RestoreTabsToBrowser [0x00007FFFF94D2A1E+382] (C:\work\cr127\brave-browser\src\chrome\browser\sessions\session_restore.cc:856)
        SessionRestoreImpl::ProcessSessionWindows [0x00007FFFF94D1F85+901] (C:\work\cr127\brave-browser\src\chrome\browser\sessions\session_restore.cc:737)
        SessionRestoreImpl::ProcessSessionWindowsAndNotify [0x00007FFFF94D01D8+88] (C:\work\cr127\brave-browser\src\chrome\browser\sessions\session_restore.cc:571)
        SessionRestoreImpl::Restore [0x00007FFFF94CD92E+302] (C:\work\cr127\brave-browser\src\chrome\browser\sessions\session_restore.cc:274)
        SessionRestore::RestoreSession [0x00007FFFF94CD32F+319] (C:\work\cr127\brave-browser\src\chrome\browser\sessions\session_restore.cc:1226)
```

Reply:
The negative values were introduced here in order to avoid conflicts with upstream types https://github.com/brave/brave-core/pull/18649/files#diff-6733dd1524a6ec4bf9cc4e35f241285ad3e5b4175fdf0d0f17146b6bf5ebf47fR16-R18

Reply:
Can I just make these positive values far above the range of the upstream values? Is there any downside to that (other than a potential conflict at some point in the future, which will be obvious at compilation-time anyway)?

Reply:
Fixed the patch and made it so it only skips the `DCHECK` when the value is negative 955be0746b2d91f075d38bec7fe11880ad030a02